core/ir/decode_shared.c

```diff
@@ -179,17 +179,17 @@ int sve_veclen;
 int sve_veclens[] = { 128,  256,  384,  512,  640,  768,  896,  1024,
                       1152, 1280, 1408, 1536, 1664, 1792, 1920, 2048 };
 
-void
+bool
 dr_set_sve_vector_length(int vl)
 {
     /* TODO i#3044: Vector length will be read from h/w when running on SVE. */
```

Comment:
I guess we've had different philosophies about CLIENT_ASSERT over time.  CLIENT_ASSERT is debug-only and we used it early on to avoid release-build branch overhead when checking for clearly-invalid parameters like passing NULL for a `drcontext` parameter.  In "regular" code, asserts would never be used for external inputs, such as any parameter to an API function.  We decided DR was very performance-focused and we'd live with "worse" development practices to avoid such checks in all the API functions.  I guess I have mixed feelings about that and whether it was a good decision.  So for a function like `dr_set_sve_vector_length` which is probably called very rarely and has negligible performance impact, I would prefer to use stronger dev practices and have release-build handling of all external inputs.

How about a compromise: I'll add an ASSERT_CURIOSITY so we can visibly detect if someone is passing something weird, but we still have robust code with user-controlled error handling.

Reply:
Done, added ASSERT_CURIOSITY.